hip/gen_probe_base.rb

```diff
@@ -0,0 +1,59 @@
+$tracepoint_lambda = lambda { |provider, c, dir|
+  puts <<EOF
+TRACEPOINT_EVENT(
+  #{provider},
+  #{c.name}_#{SUFFIXES[dir]},
+  TP_ARGS(
+EOF
+  print "    "
+  if (c.parameters.nil? || c.parameters.empty?) && !(c.has_return_type? && dir == :stop)
+    print "void"
+  else
+    params = []
+    params.concat c.parameters.collect { |p|
+      "#{p.type}, #{p.name}"
+    } unless c.parameters.nil? || c.parameters.empty?
```

Comment:
Same code simplification if `c.parameter` is `empty?` and not `nil?` by default

Reply:
It is not, see above.